js/tests/gatecrashers_test.mjs

```diff
@@ -0,0 +1,139 @@
+import { once } from 'node:events'
+import * as cp from 'node:child_process'
+import { mkdir, writeFile, chmod } from 'fs/promises'
+
+export const tests = []
+const fetch = _fetch // to redefine the real fetch
+
+const port = 5000
+
+export const setup = async ({}) => {
+  const dir = '.'
+
+  await mkdir(`${dir}/guests`, { recursive: true })
+
+  const createFilesIn = ({ files, dirPath }) =>
+    Promise.all(
+      files.map(([fileName, content]) =>
+        writeFile(`${dirPath}/${fileName}`, JSON.stringify(content)),
+      ),
+    )
+
+  const sendRequest = async (path, options) => {
+    const response = await fetch(`http://localhost:${port}${path}`, options)
+    const { status, statusText, ok } = response
+    const headers = Object.fromEntries(response.headers)
+    let body = ''
+    try {
+      body = await response.json()
+    } catch (err) {
+      body = err
+    }
+    return { status, body, headers }
+  }
+
+  return { tmpPath: dir, createFilesIn, sendRequest }
+}
+
+// Test the server is running and writes the port in stdout
+tests.push(async ({ path, ctx }) => {
+  ctx.server = cp.spawn('node', [`${path}`])
+  const message = await Promise.race([
+    once(ctx.server.stdout, 'data'),
+    Promise.race([
+      once(ctx.server.stderr, 'data').then(String).then(Error),
+      once(ctx.server, 'error'),
+    ]).then(x => Promise.reject(x)),
+  ])
+  return message[0].toString().includes(port)
+})
+
+tests.push(async ({ eq, ctx }) => {
```

Comment:
I feel it would be better to check also if the `POST` request has successfully created the file with the proper filename and content

Reply:
done, still it will be nice to see how the checks are done in the previous exercise, also the students will probably reuse their server from previous exercise as well, meaning it should have an already tested behavior